src/tsconfig.rs

```diff
@@ -43,28 +39,35 @@ pub struct ProjectReference {
     pub tsconfig: Option<Arc<TsConfig>>,
 }
 
-#[derive(Debug, Default, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct CompilerOptions {
-    base_url: Option<PathBuf>,
-    paths: Option<FxIndexMap<String, Vec<String>>>,
-    #[serde(skip)]
-    paths_base: PathBuf,
-}
-
 impl TsConfig {
     pub fn parse(path: &Path, json: &mut str) -> Result<Self, serde_json::Error> {
         _ = json_strip_comments::strip(json);
-        let mut tsconfig: Self = serde_json::from_str(json)?;
-        tsconfig.path = path.to_path_buf();
+
+        let data: TsConfigJson = serde_json::from_str(json)?;
+
+        let mut tsconfig = Self {
+            path: path.to_path_buf(),
+            paths_base: PathBuf::new(),
+            references: data.references.as_ref().map_or_else(Vec::new, |refs| {
+                refs.iter()
+                    .map(|pr| ProjectReference { path: pr.path.clone(), tsconfig: None })
+                    .collect()
+            }),
+            data,
+        };
         let directory = tsconfig.directory().to_path_buf();
-        if let Some(base_url) = tsconfig.compiler_options.base_url {
-            tsconfig.compiler_options.base_url = Some(directory.normalize_with(base_url));
-        }
-        if tsconfig.compiler_options.paths.is_some() {
-            tsconfig.compiler_options.paths_base =
-                tsconfig.compiler_options.base_url.as_ref().map_or(directory, Clone::clone);
+
+        if let Some(compiler_options) = &mut tsconfig.data.compiler_options {
```

Comment:
Compiler options are `Option` in the upstream struct, so we have to add if conditionals everywhere to handle most of these changes.